sdk/core/core/test/cmocka/test_span.c

```diff
@@ -11,6 +11,61 @@
 
 #include <_az_cfg.h>
 
+void az_span_from_string_non_ascii()
+{
+  az_span span = AZ_SPAN_FROM_STR("12汉字345");
+  assert_int_equal(11, az_span_length(span));
+}
+
+void az_span_from_string_non_ascii_latin()
+{
+  az_span span = AZ_SPAN_FROM_STR("è");
+  assert_int_equal(2, az_span_length(span));
+}
+
+void az_span_from_string_non_ascii_roundtrip()
+{
+  az_span span = AZ_SPAN_FROM_STR("12汉字345");
+  char label2[20] = "abcdefghijklmnopq";
+  char* labelPtr = label2;
+
+  assert_int_equal(AZ_OK, az_span_to_str(labelPtr, 20, span));
+  assert_int_equal(0x31, label2[0]); // '1'
+  assert_int_equal(0x32, label2[1]); // '2'
+  assert_int_equal(0x3F, label2[2]); // '?' - this is unexpected
```

Comment:
@JeffreyRichter, @gilbertw - this is the round-tripping oddity where we aren't getting back the expected UTF-8 bytes back.

Reply:
So, we are not currently doing any byte formatting on az_span_to_str.  All we do is 
```c
memmove((void*)s, (void const*)az_span_ptr(span), span_length);
```
And then we add a 0 at the end. So the str is not expected to be formatted different than how the span was pointing to.

If you want this tests to pass, you would need to do something like:
```c
void az_span_from_string_non_ascii_roundtrip()
{
  az_span span = AZ_SPAN_FROM_STR("12汉字345");
  char label2[20] = "abcdefghijklmnopq";

  assert_int_equal(AZ_OK, az_span_to_str(label2, 20, span));
  assert_int_equal(0x31, label2[0]); // '1'
  assert_int_equal(0x32, label2[1]); // '2'
  assert_int_equal(0xffffffffffffffe6, label2[2]); // '?' - this is unexpected
  assert_int_equal(0xffffffffffffffb1, label2[3]); // '?' - this is unexpected
  assert_int_equal(0xffffffffffffff89, label2[4]); // '3'
  assert_int_equal(0xffffffffffffffe5, label2[5]); // '4'
  assert_int_equal(0xffffffffffffffad, label2[6]); // '5'
  assert_int_equal(0xffffffffffffff97, label2[7]);
  assert_int_equal(0x33, label2[8]); // 'i'
  assert_int_equal(0x34, label2[9]);
  assert_int_equal(0x35, label2[10]);
  assert_int_equal(0, label2[11]);
  assert_int_equal(0x6d, label2[12]);
  assert_int_equal(0x6e, label2[13]);
  assert_int_equal(0x6f, label2[14]);
  assert_int_equal(0x70, label2[15]);
  assert_int_equal(0x71, label2[16]); // 'q'
  assert_int_equal(0, label2[17]);
}
```
however, I would better just validate that the memory is the same with
```c
assert_memory_equal(label2, az_span_ptr(span), az_span_length(span));
// then maybe check that string was 0-terminated
assert_int_equal(0, label2[az_span_length(span)]);
```
With those 2 asserts you don't need to match each byte by byte (you might need to check with unsigned int otherwise for byte comparison)